### PR TITLE
bugfix/19794-overheatprotector-redraw

### PR DESCRIPTION
**Pull request by hubertkozik**
Added new option `chart.axisLayoutRuns`. It allows more predictable axis tick and label layout at the cost of performance. See #19794.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205674704253590

### CODE REVIEW

File: ts/Core/Chart/Chart.ts [hunk 1]
```diff
@@ -2445,30 +2443,39 @@ class Chart {
...
 
         // If the plot area size has changed significantly, calculate tick
         // positions again
-        const redoHorizontal = tempWidth / chart.plotWidth > 1.1,
-            // Height is more sensitive, use lower threshold
-            redoVertical = tempHeight / chart.plotHeight > 1.05;
-
-        if (redoHorizontal || redoVertical) {
+        let redoHorizontal = tempWidth !== chart.plotWidth,
+            redoVertical = tempHeight !== chart.plotHeight,
```
Comment (TorsteinHonsi) [inline ts/Core/Chart/Chart.ts:2456]: A notable change here is that previously we accepted a failure margin (> 1.1, > 1.05 respectively). I think we should keep that acceptance margin on these lines, but if the `overheatProtector` is > 1, lines 2476 and 2477 come into play, and those require an exact match.

File: ts/Core/Chart/ChartDefaults.ts [hunk 2]
```diff
@@ -426,6 +426,21 @@ const ChartDefaults: ChartOptions = {
...
+     * intervals. Use this property if the axes change after you redraw the
+     * chart.
+     *
+     * **Note:** Changing that option to higher than the default might decrease
+     * performance significantly, especially with bigger sets of data.
+     *
+     * @type      {number}
+     * @default   1
+     * @since     @next
+     * @apioption chart.overheatProtector
```
Comment (TorsteinHonsi) [inline ts/Core/Chart/ChartDefaults.ts:441]: We need a more descriptive name. Like `chart.axisFittingRuns`? Any better idea?

Comment (KacperMadej) [reply ts/Core/Chart/ChartDefaults.ts:441]: Some other ideas: `initFittingMaxIterations`, `axesLayoutLoopLimit`, `maxReflowTries`.

Comment (hubertkozik) [reply ts/Core/Chart/ChartDefaults.ts]: What do you think about `axesLayoutRuns`?

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**